### PR TITLE
added figure and subplot_adjust

### PR DESCRIPTION
**Pull request by Bradley-Butcher**
Trying to stub plt.figure and plt.subplots_adjust.

Uncertain about line 77:
`    **kwargs: Optional[Any]) -> Figure:
`

### CODE REVIEW

**Comment by Bradley-Butcher** [comment]
Hold up didn't realize you need multiple overload definitions

File: matplotlib-stubs/pyplot.pyi [hunk 1]
```diff
@@ -64,6 +64,25 @@ def subplots(
     dpi: int = ...,
     figsize: Tuple[float, float] = ...,
 ) -> Tuple[Figure, List[List[Axes]]]: ...
+def figure(
+    num: Optional[Union[int, str]] = ...,
+    figsize: Optional[List[float]] = ...,
```
Comment (thomkeh) [inline matplotlib-stubs/pyplot.pyi]: Maybe do it like this instead?

```suggestion
    figsize: Optional[Tuple[float, float]] = ...,
```

But I don't really mind either way.

File: matplotlib-stubs/pyplot.pyi [hunk 2]
```diff
@@ -64,6 +64,25 @@ def subplots(
...
+    dpi: Optional[int] = ...,
+    facecolor: Optional[str] = ...,
+    edgecolor: Optional[str] = ...,
+    frameon: Optional[bool] = ...,
```
Comment (thomkeh) [inline matplotlib-stubs/pyplot.pyi]: I think looking at the documentation, that this cannot be `None`.

```suggestion
    frameon: bool = ...,
```

File: matplotlib-stubs/pyplot.pyi [hunk 3]
```diff
@@ -64,6 +64,25 @@ def subplots(
...
+    FigureClass: Type[Figure] = ...,
+    clear: Optional[bool] = ...,
```
Comment (thomkeh) [inline matplotlib-stubs/pyplot.pyi]: Same here

```suggestion
    clear: bool = ...,
```

**Comment by thomkeh** [comment]
I'll just do the small changes and merge it.

**Comment by thomkeh** [comment]
The documentation said that `frameon` and `clear` are "optional", but they mean it in the sense that these parameters are not required; and not in the sense that these parameters can be `None`.

**Pull request merged**